### DISCUSSION TITLE
Interlisp in a terminal, how did it work?

### DISCUSSION THREAD

**Discussion by lispm** [Interlisp/medley#1215] [General]
The more modern Interlisp uses a GUI,  but the older versions must have used the terminal for the development environment. What did it look like? How usable was it? What kind of features did it offer to make terminal use comfortable?

Is there any media (videos, ...) which shows how to use Interlisp in a terminal?

Are there any versions of Interlisp which can be used inside a typical UNIX/Linux terminal with IDE support.

**Comment by Kirtai** [comment]
I think that would be Interlisp-10 while Medley is Interlisp-D?
Lars Brinkhoff has some stuff in a repo [here](https://github.com/PDP-10/Interlisp-10) that might be useful.

**Comment by nbriggs** [comment]
The TTY-oriented editor is still present in Medley/Interlisp-D.   In an Interlisp exec window, do `(EDITMODE 'TELETYPE)` and then when you edit a function or variable definition it will put you in the teletype mode editor.  Ignore all the other windows around it and you'll have more of a sense for how it was - minus the clattering of the teletype printer and the piles of paper.  It's documented in various places, try chapter 17.1 of the 1986 Interlisp Reference Manual.

**Comment by Anzus** [comment]
If you want to use a terminal-based Interlisp there’s also Blake McBride’s LISPF4 (https://github.com/blakemcbride/LISPF4). I think twenex.org <http://twenex.org/> might also have Interlisp-10?

**Comment by larsbrinkhoff** [comment]
Yes, I do believe twenex.org has Interlisp-10 installed.  Perhaps also
the Panda TOPS-20 distribution?

I believe no one has yet attempted to build Interlisp-10 from the source
files that are available in the repository that was mentioned.

**Comment by masinter** [comment]
Interlisp/VAX? Interlisp/360? are also mentioned.
On request from  @ecraven I recorded some sessions using mainly EDITMODE(TELETYPE) but I don't think they're very useful since I don't think I recorded keystrokes. I'm willing to do more -- I wonder if there's a way to automatically record keystrokes as well.

**Comment by nbriggs** [comment]
@masinter - if you turn on the accessibility keyboard, and display in light mode, then you can see what keys are being typed in the keyboard viewer... here I'm holding shift-ctrl-command and about to hit 4 to grab the piece of the screen you see here.  If you're screen recording, and have the screen real-estate that this doesn't mess it all up, it might work well enough.  You can also change the opacity of the viewer and overlap something that you need to see but not that well...

![image](https://github.com/Interlisp/medley/assets/8138841/bbee0889-e126-4b9e-8d80-1fcedb0b60db)

**Reply by nbriggs** [reply]
... though it is only going to work for the keys that are present in the base keyboard hardware - you can't get it to show you, for instance, the keyboard image for a full numeric keypad USB keyboard plugged into a laptop.

**Reply by masinter** [reply]
I think you might have replied to the wrong issue? This one was about running Interlisp-10 or EDITMODE(TELETYPE) or WINDOWWORLD(NIL) or CHATSERVER.

**Comment by nbriggs** [comment]
@masinter "I wonder if there's a way to automatically record keystrokes as well" - you can get the keystrokes visible in the video with the keyboard viewer.  If you want it machine readable, use a dribble file while you're editing with the teletype editor?
